octobot_trading/personal_data/orders/order.py

```diff
@@ -385,7 +388,7 @@ async def on_open(self, force_open=False, is_from_exchange_data=False):
 
     async def on_fill(self, force_fill=False, is_from_exchange_data=False):
         logging.get_logger(self.get_logger_name()).debug(f"on_fill triggered for {self}")
-        if self.is_open() and not self.is_refreshing():
+        if (self.is_open() and not self.is_refreshing()) or self.is_pending_creation():
```

Comment:
Should we also check is_refreshing ?

Reply:
in the particular case of `is_pending_creation`,  we are in the sync from exchange call of pending creation orders. that means that it's still refreshing but we should proceed anyway. In other words, we skip the open state of the order and we go directly to filled (from pending creation)